os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -111,9 +112,19 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 
 		mm_takesemaphore(heap);
 
-		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+		prev = NULL;
+		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; prev = node, node = (struct mm_allocnode_s *)((char *)node + node->size)) {
 			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
 
+			if (prev && (prev->size != (node->preceding & ~MM_ALLOC_BIT))) {
```

Comment:
How about moving line 117 under this checking?
Because from line 119 to 126 means there is a corruption, then we don't need to show the current node's information.

Reply:
Line 117 was present in the old code. I assumed it is for obtaining detailed logs of all nodes in the heap. If we move it under the if condition, then it will allow us to get the logs only for the corrupted node case. I think we can leave it at current position.